src/Twig/EntryFilesTwigExtension.php

```diff
@@ -18,10 +18,12 @@
 final class EntryFilesTwigExtension extends AbstractExtension
 {
     private $container;
+    private $strictMode;
 
-    public function __construct(ContainerInterface $container)
+    public function __construct(ContainerInterface $container, bool $strictMode)
```

Comment:
I think this needs to be an argument on `EntrypointLookup` - that's where the exception originates - and strict mode should also apply if someone were using the service directly for some reason.

Reply:
Yeah, I tried to do that at first iteration. I required $strictMode in EntrypointLookup and it turned out into patching almost every method with the strict mode logic and still it continued to throw exception in `EntrypointLookupCollection::getEntrypointLookup` because it must return `EntrypointLookupInterface` but there was no result there. I started to think whether I should change its signature to allow null or return a EntrypointLookupInterface stub there. Then I gave up that and ended up with a more elegant solution - just catch exceptions in 5 twig functions and that's it.
Here is what the original solution looked like (left side):
![image](https://user-images.githubusercontent.com/1675033/54992169-57b3d400-4fc7-11e9-8c0f-01552ef4b95f.png)

Reply:
Hmm. I understand. But I still think we need to do it the uglier way, because if we merge it how it is now, someone will eventually file a bug that strict mode only works in Twig, and not directly with the services (and we do support using the services directly).

Your patch in `EntrypointLookup` makes sense to me - if the file doesn't exist, just pretend like it's a file full nothing - no entries.

> still it continued to throw exception in EntrypointLookupCollection::getEntrypointLookup because it must return EntrypointLookupInterface but there was no result there

I don't quite understand this part. Even if the `entrypoints.json` file doesn't exist, the `EntrypointLookupCollection` should still be configured to know which "build names" are valid and should still be populated with those `EntrypointLookup` objects. Unless your code is trying to find a build called `foo_bar`, and that's not configured under the `builds` section for this bundle. But that *should* be invalid. I think I'm missing something here... Let me know :)

Reply:
I see your point - `EntrypointLookupCollection::getEntrypointLookup` fails because the build is not configured but not because the build file doesn't exist. That is, I modified test `$kernel->builds = ['different_build' => 'missing_build'];` so now it knows about `different_build` and tests pass.
I moved the entire logic to the entrypoint lookup classes instead of twig extension.

Reply:
@weaverryan makes sense?